src/changelogs/tags.py

```diff
@@ -0,0 +1,342 @@
+            'Heavy melee': 'Heavy Melee',
+            'melee': 'Melee',
+            'melees': 'Melee',
+            'pause': 'Pause',
+            'pauses': 'Pause',
+            'paused': 'Pause',
+            'vaults': "Sinner's Sacrifice",
+            'vault': "Sinner's Sacrifice",
+            'Vault': "Sinner's Sacrifice",
+            "Sinner's sacrifice": "Sinner's Sacrifice",
+            'Sinners sacrifice': "Sinner's Sacrifice",
+            "Sinners' sacrifice": "Sinner's Sacrifice",
+            'Sinner sacrifice': "Sinner's Sacrifice",
+            "sinner's sacrifice": "Sinner's Sacrifice",
+            'sinners sacrifice': "Sinner's Sacrifice",
+            "sinners' sacrifice": "Sinner's Sacrifice",
+            'sinner sacrifice': "Sinner's Sacrifice",
+            'flex slot': 'Flex Slot',
+            'Flex slot': 'Flex Slot',
+            'flex slots': 'Flex Slot',
+            'Flex': 'Flex Slot',
+            'guns': 'Gun',
+            'gun': 'Gun',
+            'Guns': 'Gun',
+            'Pits': 'Pit',
+            'pit': 'Pit',
+            'pits': 'Pit',
+        }
+
+        # Relations between a child and parent tag.
+        # i.e. Abrams is a parent to Siphon Life, and a child to Hero
+        # tags below are after _remap_tag() is called
+        # key = parent
+        # value = children
+        # also see self.parent_lookup
+        self.tag_tree = {
+            'Item': {'Weapon Item': {}, 'Vitality Item': {}, 'Spirit Item': {}},
+            'Hero': {'Ability': {}, 'Gun': {}},
+            'Melee': {
+                'Light Melee': {},
+                'Heavy Melee': {},
+                'Parry': {},
+            },
+            'NPC': {'Denizen': {}, 'Creep': {'Denizen': {}, 'Trooper': {}}, 'Mid-Boss': {}},
+            'Objective': {
+                'Guardian': {},
+                'Base Guardian': {},
+                'Walker': {},
+                'Patron': {'Weakened Patron': {}},
+                'Shrine': {},
+            },
+            'Souls': {'Soul Orb': {}, "Sinner's Sacrifice": {}, 'Soul Urn': {}},
+            'Breakable': {'Crate': {}, 'Golden Statue': {}},
+            'Shop': {},
+            'Map': {
+                'Department Store': {},
+                'Arcade': {},
+                'Fish Market': {},
+                'Basement': {},
+                'Bridge': {},
+                'Hotel': {},
+                'Pharmacy': {},
+                'Penn Station': {},
+                'Grand Central': {},
+                'Pit': {},
+            },
+            'Powerup': {},
+            'Zipline': {},
+            'Rejuvenator': {},
+            'Cosmic Veil': {},
+            'Sapphire Hand': {},
+            'Amber Hand': {},
+            'Sandbox': {},
+            'Pause': {},
+            'Bounce Pad': {},
+            'Rope': {},
+            'Flex Slot': {},
+            'Greenwich': {},
+            'York Avenue': {},
+            'Broadway': {},
+            'Park Avenue': {},
```

Comment:
should the lane names be here?

Reply:
Don't see a reason not to include em

Reply:
Would it fit into the "Map" parent?

Reply:
I thought about this too; I added it then thought about adding Bounce Pads, Rope, Zipline, etc. But then I realized that if these are referenced in gameplay changes, then its nothing to do with the map, so I removed the lane names too. Though thinking again now, I think your right; the lane names can go into Map because they should only be referenced in map-related changes, similar to the other map locations.

Reply:
Its definitely a difficult thing to figure out, since it depends how the wiki is structured, which can change a lot. 

Possibly we could consider excluding parent tags from the changelogs themselves, and have a lookup table to find any parents? Not one for this PR, but it would mean these decisions are less crucial, since a change to the tag tree wouldn't require reparsing the changelogs

Reply:
It also means tagging changelogs manually is simpler. Eg. if an item change is missing its tag, you don't need to add the item name, "Item", and "Weapon Item", you just need to add the item name

Reply:
Not quite, it's more to do with removing any redundant tags from the changelog files. Eg. if a changelog contains "Weakened Patron", it wouldn't need to have "Patron" or "Objective" explicitly written in the tags list, as that can be defined by a child-parent map stored separate

Reply:
But its only a light suggestion, I'm not sure thats so great, as having them baked in keeps the wiki simpler

Reply:
Hmm, I *think* I understand the idea. But yeah this could be a route for improvement down the line. Luckily, after the `tags.py` is configured, there isn't much that needs to be maintained with it, as in games like this 90% of new content is new resources (i.e. new heroes, items), as opposed to new resource types or mechanics which would require more tags to be handled explicitly.